resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -418,17 +417,33 @@ private[spark] class MesosClusterScheduler(
     envBuilder.build()
   }
 
+  private def isContainerLocalAppJar(desc: MesosDriverDescription): Boolean = {
+    val isLocalJar = desc.jarUrl.startsWith("local://")
+    val isContainerLocal = desc.conf.getOption("spark.mesos.appJar.local.resolution.mode").exists {
+      case "container" => true
+      case "host" => false
+      case other =>
+        logWarning(s"Unknown spark.mesos.appJar.local.resolution.mode $other, using host.")
+        false
+      }
```

Comment:
Can we do:

```scala
desc.conf.getOption("spark.mesos.appJar.local.resolution.mode") match {
    case Some("container") => true
    case Some("host") | None => false
    case Some(other) =>
        ...
}
```
?

Reply:
What's wrong with exists?

Reply:
Nothing wrong but I just find it hard to read. I assume @felixcheung had a similar concern at the core.

Reply:
It's not very commonly used, but ok to me.

Reply:
Ok I see.